tests/net_dynamic_hb.rs

```diff
@@ -169,7 +169,27 @@ fn do_drop_and_readd(cfg: TestConfig) {
     // Run the network:
     loop {
         let (node_id, step) = net.crank_expect();
-
+        if !net[node_id].is_faulty() {
+            // Verify that contributions from all participants of the batch are present.
+            for batch in &step.output {
+                let expected_participants: Vec<_> = if awaiting_removal.contains(&node_id) {
+                    // The node hasn't removed the pivot node yet.
+                    pub_keys_add.keys()
+                } else if awaiting_addition.contains(&node_id) {
+                    // The node has removed the pivot node but hasn't added it back yet.
+                    pub_keys_rm.keys()
+                } else {
+                    // The node has added the pivot node back.
+                    pub_keys_add.keys()
+                }.collect();
+                let batch_participants: Vec<_> = batch.contributions().map(|(id, _)| id).collect();
+                assert_eq!(
+                    batch_participants, expected_participants,
```

Comment:
That's not something we can expect. (If it's always true in the test, that's because the test doesn't simulate any difficult network conditions.)
The guarantee is just that at least _N - f_ participants managed to get their contribution accepted.

Reply:
Should we expect only correct nodes to be batch participants in this case?

Reply:
> But at this point in the code, some of them may still be waiting for it.

Right, this place in the code is better for making sanity checks on batches, possibly checking for participants that _shouldn't_ be in the batch.

The fact that all expected participants in a batch receive it is better checked at the end.